File: archive/ruby/muse/algorithms/sorting/SelectionSortTest.rb
require "muse/algorithms/sorting/SelectionSort"
require "muse/util/SequenceBuilder"
require "muse/util/Sequences"
require "muse/util/TestRunner"

def testSelectionSort
  size = 32768

  arr = Array.new(size)
  SequenceBuilder.packRandom(arr)

  checksum = Sequences.parityChecksum(arr)

  SelectionSort.sort(arr)

  return false if Sequences.parityChecksum(arr) != checksum

  Sequences.isSorted(arr)
end

TestRunner.pick(testSelectionSort) if __FILE__ == $PROGRAM_NAME

File: archive/c++/muse/algorithms/sorting/SelectionSortTest.cpp
#include <muse/algorithms/sorting/SelectionSort.hpp>
#include <muse/util/SequenceBuilder.hpp>
#include <muse/util/Sequences.hpp>
#include <muse/util/TestRunner.hpp>

using namespace std;

bool testSelectionSort() {
  size_t size = 32768;

  vector<int> arr(size);
  SequenceBuilder::packRandom(arr);

  int checksum = Sequences::parityChecksum(arr);

  SelectionSort::sort(arr);

  if (Sequences::parityChecksum(arr) != checksum) {
    return false;
  }

  return Sequences::isSorted(arr);
}

int main() { TestRunner::pick(&testSelectionSort); }
